File: mvp-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/userdetails/UserDetailsView.java
package com.wasisto.githubuserfinder.android.ui.userdetails;

import com.wasisto.githubuserfinder.domain.models.User;

public interface UserDetailsView {

    void showLoadingIndicator();

    void hideLoadingIndicator();

    void showUser(User user);

    void openBrowser(String url);

    void showErrorToast();

    void closeActivity();
}

File: mvp-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/userdetails/UserDetailsView.kt
package com.wasisto.githubuserfinder.android.ui.userdetails

import com.wasisto.githubuserfinder.domain.models.User

interface UserDetailsView {

    fun showLoadingIndicator()

    fun hideLoadingIndicator()

    fun showUser(user: User)

    fun openBrowser(url: String)

    fun showErrorToast()

    fun closeActivity()
}